Terminal.Gui/Views/Menu/MenuBar.cs

```diff
@@ -0,0 +1,1471 @@
+/// </para>
+/// </remarks>
+public class MenuBar : View {
+	internal int _selected;
+	internal int _selectedSub;
+
+	/// <summary>
+	/// Gets or sets the array of <see cref="MenuBarItem"/>s for the menu. Only set this after the <see cref="MenuBar"/> is visible.
+	/// </summary>
+	/// <value>The menu array.</value>
+	public MenuBarItem [] Menus { get; set; }
+
+	/// <summary>
+	/// The default <see cref="LineStyle"/> for <see cref="Menus"/>'s border. The default is <see cref="LineStyle.Single"/>.
+	/// </summary>
+	public LineStyle MenusBorderStyle { get; set; } = LineStyle.Single;
+
+	bool _useSubMenusSingleFrame;
+
+	/// <summary>
+	/// Gets or sets if the sub-menus must be displayed in a single or multiple frames.
+	/// <para>
+	/// By default any sub-sub-menus (sub-menus of the main <see cref="MenuItem"/>s) are displayed in a cascading manner, 
+	/// where each sub-sub-menu pops out of the sub-menu frame
+	/// (either to the right or left, depending on where the sub-menu is relative to the edge of the screen). By setting
+	/// <see cref="UseSubMenusSingleFrame"/> to <see langword="true"/>, this behavior can be changed such that all sub-sub-menus are
+	/// drawn within a single frame below the MenuBar.
+	/// </para>		
+	/// </summary>
+	public bool UseSubMenusSingleFrame {
+		get => _useSubMenusSingleFrame;
+		set {
+			_useSubMenusSingleFrame = value;
+			if (value && UseKeysUpDownAsKeysLeftRight) {
+				_useKeysUpDownAsKeysLeftRight = false;
+				SetNeedsDisplay ();
+			}
+		}
+	}
+
+	///<inheritdoc/>
+	public override bool Visible {
+		get => base.Visible;
+		set {
+			base.Visible = value;
+			if (!value) {
+				CloseAllMenus ();
+			}
+		}
+	}
+
+	/// <summary>
+	/// Initializes a new instance of the <see cref="MenuBar"/>.
+	/// </summary>
+	public MenuBar () : this (new MenuBarItem [] { }) { }
+
+	/// <summary>
+	/// Initializes a new instance of the <see cref="MenuBar"/> class with the specified set of Toplevel menu items.
+	/// </summary>
+	/// <param name="menus">Individual menu items; a null item will result in a separator being drawn.</param>
+	public MenuBar (MenuBarItem [] menus) : base ()
+	{
+		X = 0;
+		Y = 0;
+		Width = Dim.Fill ();
+		Height = 1;
+		Menus = menus;
+		//CanFocus = true;
+		_selected = -1;
+		_selectedSub = -1;
+		ColorScheme = Colors.Menu;
+		WantMousePositionReports = true;
+		IsMenuOpen = false;
+
+		Added += MenuBar_Added;
+
+		// Things this view knows how to do
+		AddCommand (Command.Left, () => {
+			MoveLeft ();
+			return true;
+		});
+		AddCommand (Command.Right, () => {
+			MoveRight ();
+			return true;
+		});
+		AddCommand (Command.Cancel, () => {
+			CloseMenuBar ();
+			return true;
+		});
+		AddCommand (Command.Accept, () => {
+			ProcessMenu (_selected, Menus [_selected]);
+			return true;
+		});
+
+		AddCommand (Command.ToggleExpandCollapse, () => SelectOrRun ());
+		AddCommand (Command.Select, () => SelectOrRun ());
+		AddCommand (Command.Select, () => Run (_menuItemToSelect?.Action));
```

Comment:
You are specifying 2 definitions for this command.  This will just overwrite the implementation and you will end up only with line 307 taking effect